HashRegistrarSimplified.sol

```diff
@@ -461,6 +468,36 @@ contract Registrar {
     }
 
     /**
+     * @dev Allows anyone to delete the owner and resolver records for a (subdomain of) a
+     *      name that is not currently owned in the registrar. If passing, eg, 'foo.bar.eth',
+     *      the owner and resolver fields on 'foo.bar.eth' and 'bar.eth' will all be cleared.
+     * @param labels A series of label hashes identifying the name to zero out, rooted at the
+     *        registrar's root. Must contain at least one element. For instance, to zero 
+     *        'foo.bar.eth' on a registrar that owns '.eth', pass an array containing
+     *        [sha3('foo'), sha3('bar')].
+     */
+    function eraseNode(bytes32[] labels) {
+        if(labels.length == 0) throw;
+        if(state(labels[labels.length - 1]) == Mode.Owned) throw;
+
+        _eraseNode(labels.length - 1, labels, rootNode);
+    }
+
+    function _eraseNode(uint idx, bytes32[] labels, bytes32 node) internal {
+        // Take ownership of the node
+        ens.setSubnodeOwner(node, labels[idx], address(this));
+        node = sha3(node, labels[idx]);
+        
+        // Recurse if there's more labels
+        if(idx > 0)
+            _eraseNode(idx - 1, labels, node);
```

Comment:
I did consider that; frankly, I'm happy for you to keep your ridiculously long domain, even if the 2LD is 'too short'.

There's no way I'm aware of to shortcut this without adding lots of complexity (such as storing intermediate values in contract storage).